### PR TITLE
Try cascading nav styles through classnames

### PR DESCRIPTION
**Pull request by tellthemachines**
## Description

Alternative to #37438. Instead of adding custom properties to layout, classnames are added to the navigation block wrapper and custom properties are hooked to those classnames. 

I think this is neater than adding the custom properties as inline styles, because this way they live in a single place - the stylesheet - instead of being duplicated across JS and PHP files. Also, we're already generating a couple of the justification classnames anyway, to attach submenu styles. 

## How has this been tested?

Create a Navigation block; change justification/orientation and verify everything works as expected in both editor and front end.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by jasmussen** [comment]
Thanks for the PR! To me this feels like a good pragmatic step forward. Justifications in a classname affords the same opportunities inheritance does, but a bit more flexibility and simplicity, I think it works well, and in a quick testing, justifications are working across the board for the navigation block. Nice!

Could we change the classnames from `wp-justify-[justification]` though? There are a number of blocks across the codebase that use the pattern `items-justified-[justification]`, which feels more agnostic and simple. What do you think?

File: packages/block-library/src/navigation/style.scss [hunk 1]
```diff
@@ -106,6 +106,45 @@ $navigation-icon-size: 24px;
 			height: inherit;
 		}
 	}
+
+	// Custom properties for layout style cascade: default values.
+	--layout-justification-setting: flex-start;
```
Comment (youknowriad) [inline packages/block-library/src/navigation/style.scss]: Can we make these CSS variables more specific to the navigation block or something? They feel a bit too generic.

**Review by youknowriad — COMMENTED** [review]
This is a great approach. Close to what I had in mind with the "no-filters" approach. 👍

File: packages/block-library/src/navigation/style.scss (hunk 1 above)
Comment (tellthemachines) [reply packages/block-library/src/navigation/style.scss]: Done! I prefixed them with "navigation".

**Comment by tellthemachines** [comment]
@jasmussen the classnames we're using here are the `items-justified-` ones, see [here](https://github.com/WordPress/gutenberg/blob/try/nav-classnames/packages/block-library/src/navigation/edit/index.js#L211). I hadn't noticed those `wp-justify-` classes were also being added; they seem to have been introduced recently in [this changeset](https://github.com/WordPress/gutenberg/commit/18214afc6618f95e0e724926aba5aada28a00dbc).

**Comment by jasmussen** [comment]
Sorry about that, and thanks for the pointer! I guess this one just needs a code review and a green light.

**Comment by youknowriad** [comment]
I think the layout justify classnames are also being removed cc @ntsekouras.

**Review by youknowriad — APPROVED** [review]
👍

**Comment by ntsekouras** [comment]
>I think the layout justify classnames are also being removed cc @ntsekouras.

Yes, that PR [will be reverted](https://github.com/WordPress/gutenberg/pull/36681#issuecomment-996028742).

**Pull request merged**